torchao/prototype/mx_formats/constants.py

```diff
@@ -3,7 +3,8 @@
 # This is conceptually an enum of non-core dtypes
 # TODO(future PR): change to a cleaner way to represent this without
 # regressing torch.compile and while keeping things readable.
-DTYPE_FP4 = "fp4_e2m1"
+DTYPE_FP4_E2M1 = "fp4_e2m1"
+DTYPE_FP4_E3M0 = "fp4_e3m0"
```

Comment:
since `fp4_e3m0` is not in the MX spec yet, is there any context we can add here for the inclusion of the dtype here?  Are there links to papers / studies / etc?

this PR is good to go, I'm just trying to see if we can help future code readers understand this better

Reply:
actually, can we also add that information to `torchao/prototype/mx_formats/README.md`?

Reply:
> since `fp4_e3m0` is not in the MX spec yet, is there any context we can add here for the inclusion of the dtype here? Are there links to papers / studies / etc?
> 
> this PR is good to go, I'm just trying to see if we can help future code readers understand this better

The motivation for adding E3M0 came from the Intel team's observation that the distribution of gradients approximates a log-normal distribution. They also published a [paper](https://openreview.net/pdf?id=EoFNy62JGd) about it.

Reply:
> actually, can we also add that information to `torchao/prototype/mx_formats/README.md`?

Sure! I will update the README in the following commit.